lib/host-connection-pool.js

```diff
@@ -255,49 +317,58 @@ HostConnectionPool.prototype.drainAndShutdown = function () {
     c.emitDrain = true;
     c.once('drain', getDelayedClose(c));
   }
-
-  var wasSetAsShutdown = false;
+  var wasClosed = false;
   var checkShutdownTimeout;
   function getDelayedClose(connection) {
     return (function delayedClose() {
       connection.close();
-      connection.setAsClosed = true;
       if (++closedConnections < connections.length) {
         return;
       }
-      if (wasSetAsShutdown) {
+      if (wasClosed) {
         return;
       }
-      wasSetAsShutdown = true;
+      wasClosed = true;
       if (checkShutdownTimeout) {
         clearTimeout(checkShutdownTimeout);
       }
-      self.setAsShutdown();
+      self._afterClosing();
     });
   }
-  // check that after sometime (readTimeout + 2 secs) the connections have been drained
-  var delay = (this.options.socketOptions.readTimeout || defaultOptions.socketOptions.readTimeout) + 2000;
-  setTimeout(function checkShutdown() {
-    if (wasSetAsShutdown) {
-      return;
-    }
-    wasSetAsShutdown = true;
-    self.setAsShutdown();
+  // Check that after sometime (readTimeout + 100ms) the connections have been drained
+  var delay = (this.options.socketOptions.readTimeout || defaultOptions.socketOptions.readTimeout) + 100;
```

Comment:
Is there a main reason for reducing the delay time, is it just because 2 seconds is probably too much time?